lib/queue.rb

```diff
@@ -1,31 +1,79 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(10)
+    @front = @back = 0
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    # If circular buffer is full, make a new, bigger array
+    if @front == (@back + 1) % size
+      old_queue_size = @store.length
+      new_store = Array.new(old_queue_size + 10)
```

Comment:
Interesting how you're handling the resizing.  I would suggest maybe just doubling the size of the buffer instead of adding 10.